uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js

```diff
@@ -379,13 +379,24 @@ define([
         }
 
         if (missingRequiredFields.length) {
-          var notification = CRM.alert(missingRequiredFields.join(', '),
-            missingRequiredFields.length === 1 ? 'Required field' : 'Required fields', 'error');
+          var notification;
+          var notificationTitle = missingRequiredFields.length === 1 ? 'Required field' : 'Required fields';
+          var missingFields = missingRequiredFields.join(', ');
+
+          if ($scope.isRole('manager')) {
+            $rootScope.$broadcast('ssp:notify', {
+              type: 'error',
+              title: notificationTitle,
+              body: missingFields
+            });
+          } else {
+            notification = CRM.alert(missingFields, notificationTitle);
```

Comment:
`CRM.alert()` is available in SSP. The problem is that the way it works in SSP.  Here
```js
/**
   * @see https://wiki.civicrm.org/confluence/display/CRMDOC/Notification+Reference
   */
  CRM.alert = function (text, title, type, options) {
    debugger
    type = type || 'alert';
    title = title || '';
    options = options || {};
    if ($('#crm-notification-container').length) {
      debugger
      var params = {
        text: text,
        title: title,
        type: type
      };
      // By default, don't expire errors and messages containing links
      var extra = {
        expires: (type == 'error' || text.indexOf('<a ') > -1) ? 0 : (text ? 10000 : 5000),
        unique: true
      };
      options = $.extend(extra, options);
      options.expires = options.expires === false ? 0 : parseInt(options.expires, 10);
      if (options.unique && options.unique !== '0') {
        $('#crm-notification-container .ui-notify-message').each(function () {
          if (title === $('h1', this).html() && text === $('.notify-content', this).html()) {
            $('.icon.ui-notify-close', this).click();
          }
        });
      }
      debugger
      return $('#crm-notification-container').notify('create', params, options);
    }
    else {
      if (title.length) {
        text = title + "\n" + text;
      }
      alert(text);
      return null;
    }
  };
```
The `CRM.alert()` looks for the element with id `#crm-notification-container` which is not present in SSP so it simply displays the default alert.